[andrechuakj] iP

andrechuakj opened the pull request:
# Your Chatbot
> "When you fall, I'll be there" - Floor (Source: _trust me_)
Your Chatbot frees your mind so that you have more space for important things like Djikstra algorithm. It,
- doesn't have a GUI so that you can train your typing speed 🥇 
- doesn't need a GUI because a true coder uses CLI 😎 
- uses OOP 💯 

All you need to do is,
1. Download it from 👉 [here](https://youtu.be/dQw4w9WgXcQ?si=pT5heV9Bpg1ZJyEA). 👈 
2. Double click it and pray 🙏 that the code runs.
3. Add your tasks ✅ 
4. Procrastinate 
5. ??
6. Profit 💵 

And it is **FREE**!

Other features
- [X] O(1) adding task
- [X] O(n) deleting task (will be optimized in the future)
- [X] Only has v0.1
- [ ] Bugs (coming soon)
- [ ] Funny description (work in progress)
---
If you are a Java programmer or if you are not, you can use Your Chatbot to practice Java too!! Here's the `main` method:
```java
public class Duke {
    public static void main(String[] args) throws DukeException {
        new Duke("data/duke.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

andrechuakj committed:
Renamed bot, added greet and exit

andrechuakj committed:
Added echo feature

andrechuakj committed:
Added add and list feature

andrechuakj committed:
Added mark as done

andrechuakj committed:
Added Todos, events, deadlines

andrechuakj committed:
Added testing

andrechuakj committed:
Added error handling

andrechuakj committed:
Added delete feature

andrechuakj committed:
Add save function

andrechuakj committed:
Merge branch 'branch-level-7'

andrechuakj committed:
Add dates

andrechuakj committed:
Merge branch 'branch-level-8'

andrechuakj committed:
Refactor code with Ui class

andrechuakj committed:
Refactor code with Storage class

nixonwidjaja commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,259 @@
+        }
+
+        boolean end = false;
+        while (!end) {
```
Perhaps consider abstracting your if else or switch case statements so there's not much indentation here?

andrechuakj replied:
Thanks for the feedback, I've taken that into consideration and updated my code with more abstraction.

nixonwidjaja reviewed (changes requested):
Great job on making your code simple and concise! Perhaps consider abstracting away some logics in your Duke.java into different classes to make your code more modular and readable. I'm not sure if this is your final code at this moment since I don't see any packages and Gradle is not integrated yet. Overall, good work!

andrechuakj committed:
Refactor code with Parser and Command classes

andrechuakj committed:
Merge branch 'branch-A-MoreOOP'

andrechuakj committed:
Organize into packages

andrechuakj committed:
Merge branch 'branch-A-Packages'

andrechuakj committed:
Merge branch 'add-gradle-support'

andrechuakj committed:
Add gradle support

andrechuakj committed:
Add JUnit tests for Parser and Deadline classes

andrechuakj committed:
Merge branch 'branch-A-JUnit'

andrechuakj committed:
Add JavaDocs

andrechuakj committed:
Correct coding standard

andrechuakj committed:
Add find command

andrechuakj committed:
Merge branch 'branch-A-JavaDoc'

andrechuakj committed:
Merge and resolve conflict

andrechuakj committed:
Merge and resolve conflict with branch-Level-9

ryamgoh commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,18 @@
+ * Custom exception class for Duke application.
+ * DukeException is thrown when there are errors or exceptions specific to the Duke application.
+ */
+public class DukeException extends Exception {
```
Hi Andre, code looks hella clean!

I think it will also be nice to segment your code into packages for nicer file structure

Something similar to this:
![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/122634188/760fede6-814b-417f-a278-90264955cc23)

ryamgoh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,170 @@
+            + "| |   | | (_) | |_| | |    | \\__/\\ | | | (_| | |_| |_) | (_) | |_ | |\n"
+            + "| |_  \\_/\\___/ \\__,_|_|     \\____/_| |_|\\__,_|\\__|_.__/ \\___/ \\__|| |\n"
+            + "|___|                                                           |___|\n";
+    private static final String greetPhrase = LINE
```
A final variable should be in capital case! :)

greetPhrase => GREETPHRASE

Likewise for SENDOFFPHRASE

ryamgoh commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,94 @@
+        this.taskType = taskType;
+    }
+
+    @Override
```
I think can still write a javadoc for these execute statements since they have differing behaviours across the different classes! :)

ryamgoh commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,29 @@
+import java.util.ArrayList;
+
+public class FindCommand extends Command {
+
```
javadocs for constructors? In case got overloaded constructor

andrechuakj committed:
Correct with checkstyle

andrechuakj committed:
Merge branch 'branch-A-CheckStyle'

andrechuakj committed:
Add GUI

andrechuakj committed:
Change Ui class to Reponse

andrechuakj committed:
Merge branch 'branch-Level-10'

andrechuakj committed:
Change javafx from version 20 to 17.0.7

andrechuakj committed:
Add assertions

Absence of assertions in our code makes it difficult to detect bugs
and unexpected conditions.

Assertions enhance the reliability and robustness of our Java files.
Introducing assertions at critical points will help us catch potential
issues early in the development process and make our software more
resilient.

Let's,
* add assertions in [Duke.java, Parser.java, AddCommand.java]

andrechuakj committed:
Improve code quality

There are several blocks of code that can improve on readability.

Improving readability of code allows for easier debugging and
maintenance of code.

Let's,
* Rename DeleteComand to DeleteCommand
* Response class: Add a newline between the command and the description
  in the printFind mtehod response message
* FindCommand class: Improve the readability of the for loop by
  utilizing the enhanced for loop
* MarkCommand class: Improve the readability of the happy path by
  removing the unnecessary assignment of 'taskNumber' to 0
  before parsing
* AddCommand class: Extract the creation of task into a separate
  method

andrechuakj committed:
Merge pull request #1 from andrechuakj/branch-A-Assertions

Add assertions

andrechuakj committed:
Resolve merge conflicts

andrechuakj committed:
Merge pull request #2 from andrechuakj/branch-A-CodeQuality

Improve code quality

andrechuakj committed:
Set up CI

andrechuakj committed:
Merge branch 'A-CI'

andrechuakj committed:
Remove I/O redirection tests from gradle.yml

andrechuakj committed:
Merge branch 'branch-A-CI'

andrechuakj committed:
Add update command

andrechuakj committed:
Merge branch 'branch-C-Update'

andrechuakj committed:
Add user guide

andrechuakj committed:
Merge branch 'branch-A-UserGuide'

andrechuakj committed:
Rename Ui.png.png to Ui.png

andrechuakj committed:
Edit site theme to minima

andrechuakj committed:
Parser: Extract methods from dataToTask.

andrechuakj committed:
Parser: extract methods from parse method

andrechuakj committed:
Resolve test problems and update build.gradle
